Automattic/wp-calypso
# Plans grid: billing interval Toggle component

### Changes proposed in this Pull Request

* Add the `<PlansIntervalToggle />` component to the `@automattic/plans-grid` package, but it DOESN'T render it
* This component is heavily based on the [`<SegmentedControl />` calypso component](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/segmented-control) and [how it is used in the /start flow](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/plan-type-selector.tsx)
* Compared to [how this toggle has been implemented in other flows (`/start`)](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/plan-type-selector.tsx), this implementation has a few differences:
  * the `Popover` component is imported from [`@wordpress/components`](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/popover) instead of `calypso/components/popover`. Since the new version of the component has already got its own enter transition, the `CSSTransition` component was not imported/used.
  * Since the Monthly vs Annually selection doesn't need to modify the page's URL, the `path` prop is not computed and passed to the `<SegmentedControl.Item />` component. This allowed to avoid importing the `utility-types` and `lodash` packages.
  * For translations, `@automattic/react-i18n` package was used instead of `i18n-calypso`
  * Style overrides are applied via SCSS code (instead of using `styled` components)

### Screenshots

<img width="600" src="https://user-images.githubusercontent.com/1083581/104479909-3b31c300-55c4-11eb-86e9-15d9c3a4eb0c.gif">

### Testing instructions

#### Setup

1. Open this PR's code on your local machine
2. Add the `<PlansIntervalToggle />` component to the Plans grid:
<img width="1512" alt="Screenshot 2021-01-14 at 15 25 30" src="https://user-images.githubusercontent.com/1083581/104603601-dedeaa00-567c-11eb-80d4-b1f41277ec2b.png">



#### Test in Gutenboarding

- Go through the setup steps
- Run `yarn && yarn start` in the root folder of this repo
- Visit `calyspo.localhost:3000/new` and navigate to the Plan step
- Make sure that all A/Cs from #48769 are satisfied

Fixes #48769
